### ISSUE TITLE
VM Connection Error

### ISSUE THREAD

**Issue by alimp5** [MobSF/Mobile-Security-Framework-MobSF#537]
```

## CONSOLE OUTPUT

```
[INFO] Setting up Dynamic Analysis Environment

[INFO] Starting Web Proxy

[INFO] Started Web Proxy at 192.168.56.1:1337

[INFO] Connecting to VM/Device

[INFO] WebProxy Socket is already in use
error: protocol fault (couldn't read status): Success
error: protocol fault (couldn't read status): Success

[INFO] Launching APK Main Activity

[INFO] Mounting
* daemon not running. starting it now at tcp:5037 *
* daemon started successfully *
error: device '192.168.56.101:5555' not found
[INFO] Testing Environment is Ready!

[INFO] Getting Screen Resolution
error: device '192.168.56.101:5555' not found
error: device '192.168.56.101:5555' not found

[INFO] Starting App for Dynamic Analysis
error: device '192.168.56.101:5555' not found

[2018-01-27 10:56:04]
[ERROR] Getting Screen Resolution (/root/Desktop/fata/Mobile-Security-Framework-MobSF/DynamicAnalyzer/views/android/android_dyn_shared.py, LINE 80 "[adb, "-s", get_identifier(), "shell", "dumpsys", "window"])"): Command '['/root/Desktop/fata/Mobile-Security-Framework-MobSF/DynamicAnalyzer/tools/adb/linux/adb', '-s', '192.168.56.101:5555', 'shell', 'dumpsys', 'window']' returned non-zero exit status 1

[INFO] Installing APK
[27/Jan/2018 10:56:04] "POST /GetEnv/ HTTP/1.1" 200 58
error: device '192.168.56.101:5555' not found
adb: error: failed to get feature set: device '192.168.56.101:5555' not found
- waiting for device -
[2018-01-27 10:56:04,652] - Broken pipe from ('192.168.100.23', 56056)

[INFO] ADB Started

[INFO] Waiting for 5 seconds...

[INFO] Connecting to VM/Device
unable to connect to 192.168.56.101:5555: Connection refused

```


## CONTENTS OF LOG FILES

```
[2018-01-27 10:56:04]
[ERROR] Getting Screen Resolution (/root/Desktop/fata/Mobile-Security-Framework-MobSF/DynamicAnalyzer/views/android/android_dyn_shared.py, LINE 80 "[adb, "-s", get_identifier(), "shell", "dumpsys", "window"])"): Command '['/root/Desktop/fata/Mobile-Security-Framework-MobSF/DynamicAnalyzer/tools/adb/linux/adb', '-s', '192.168.56.101:5555', 'shell', 'dumpsys', 'window']' returned non-zero exit status 1

```

**Comment by alimp5** [comment]
I can ping host-only adapter, so what's the problem !?

root@kali:~# ping 192.168.56.101
PING 192.168.56.101 (192.168.56.101) 56(84) bytes of data.
64 bytes from 192.168.56.101: icmp_seq=1 ttl=64 time=0.032 ms
64 bytes from 192.168.56.101: icmp_seq=2 ttl=64 time=0.038 ms
64 bytes from 192.168.56.101: icmp_seq=3 ttl=64 time=0.045 ms
64 bytes from 192.168.56.101: icmp_seq=4 ttl=64 time=0.049 ms
^C
--- 192.168.56.101 ping statistics ---
4 packets transmitted, 4 received, 0% packet loss, time 3077ms
rtt min/avg/max/mdev = 0.032/0.041/0.049/0.006 ms

**Comment by alimp5** [comment]
DHCP Server: off
VirtualBox: Ver 5.2.6 Debian _ 64 bit
Adapter name: vboxnet0
Snapshot:
https://www.photobox.co.uk/my/photo?album_id=5237901019&photo_id=500533071941

**Comment by ajinabraham** [comment]
Looks like adb cannot communicate to `192.168.56.101:5555`. 
Try
1. Stop MobSF, close VirtualBox
2. Kill all adb process and try connecting again.
3. Also try restarting your system and see if you can connect.
4. If the above doesn't work, see if you are able to connect to the VM using adb.

**Comment by ajinabraham** [comment]
Getting Screen Resolution Error is showing because the connection to VM itself is not working.

**Comment by alimp5** [comment]
@ajinabraham 
Hi again.
problem fixed.
I couldn't find the root cause of connection-refusing between VirtualBox and MobSF Web User Interface and MobSF-VM.

because I opened and closed and reconfigured many times the MobSF-VM and VirtualBox-Network-Host-Manager  to obtain a correct IP-Address for Dynamic-Analysis phase. And at the end I could connect MobSF to VM. 👍 💯 


But, When I clicked on Report-Generation in Dynamic-Analysis section, I saw below error in console:
[2018-01-28 13:29:12]
[ERROR] Parsing JSON Failed for: {"timestamp":1517145548299,"result":"\n0x00000000 10 66 6B 51 E6 0C 20 24 63 D4 27 91 5E DD 15 35 .fkQ...$c.'.^..5","class":"android.util.Base64","method":"decode","type":"generic","args":["EGZrUeYMICRj1CeRXt0VNQ==","0"]}
 (/root/Desktop/fata/Mobile-Security-Framework-MobSF/DynamicAnalyzer/views/android/android_dynamic.py, LINE 1006 "escape(base64.b64decode(args_list[0]))"): 'utf8' codec can't decode byte 0xe6 in position 4: invalid continuation byte. You passed in "\x10fkQ\xe6\x0c $c\xd4'\x91^\xdd\x155" (<type 'str'>)

Have great times.
